### ISSUE TITLE
Converge with CSSOM-View

### ISSUE THREAD

**Issue by garykac** [w3c/uievents#24]
Copied from W3C Bugzilla: https://www.w3.org/Bugs/Public/show_bug.cgi?id=25928

##### Anne 2014-05-30 18:27:08 UTC
Seems like it might be time to fix this weirdness: http://www.w3.org/TR/cssom-view/#extensions-to-the-mouseevent-interface

Change stuff from integers to floats, add the missing members, and define the members better.

##### Comment 1 Travis Leithead [MSFT] 2014-05-30 18:42:11 UTC
Awesome. Thanks for the head's up.

##### Comment 2 Arthur Barstow 2015-03-21 15:11:51 UTC
Bulk move of all D3E bugs to the UI Events component.

**Comment by zcorpan** [comment]
Also see
https://www.w3.org/Bugs/Public/show_bug.cgi?id=29518
and
https://www.w3.org/Bugs/Public/show_bug.cgi?id=25323#c3

**Comment by cvrebert** [comment]
See also https://github.com/w3c/csswg-drafts/commit/dc36ecd7a46b173f958dafa736a84eb9753afb7b

**Comment by saschanaz** [comment]
Do we really want to make them floats when everyone is returning integers?

**Comment by zcorpan** [comment]
They're `double` in Chromium

https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/core/events/mouse_event.idl;l=26

`long` in Gecko and WebKit

https://dxr.mozilla.org/mozilla-central/source/dom/webidl/MouseEvent.webidl?q=mouseevent.webidl&redirect_type=direct#24
https://github.com/WebKit/webkit/blob/950143da027e80924b4bb86defa8a3f21fd3fb1e/Source/WebCore/dom/MouseEvent.idl#L28

**Comment by zcorpan** [comment]
Yeah, indeed. These look like the relevant bugs for `MouseEvent`:

https://bugs.chromium.org/p/chromium/issues/detail?id=660057
https://bugs.chromium.org/p/chromium/issues/detail?id=456625

They're marked as Fixed. Was it changed back? cc @eirage

**Comment by jgraham** [comment]
So it seems like Gecko has changed to match Chromium now: https://bugzilla.mozilla.org/show_bug.cgi?id=1680669 Therefore it seems like we should resolve the conflict in favour of the CSSOM View definition (i.e. these properties are `double` rather than `long`)

**Comment by mustaqahmed** [comment]
> we should resolve the conflict in favour of the CSSOM View definition

@jgraham Did you mean updating the UIEvent spec IDL only?  Or also dropping the CSSOM-View IDL afterwards?

---

A quick note regarding the past discussion here about Blink (I just saw this discussion):
- While Blink's IDL has `MouseEvent` coordinates as double, Blink still exposes only integer values for these coordinates for compat.
- As observed [above](https://github.com/w3c/uievents/issues/24#issuecomment-717231017), `PointerEvent` coordinates are fractional.  However, `click` and `ausclick` are now `PointerEvent`s but still expose only integer coordinates like a `MouseEvent`, see [pointerevents#100](https://github.com/w3c/pointerevents/issues/100#issuecomment-818870316).

**Comment by jgraham** [comment]
> Did you mean updating the UIEvent spec IDL only? Or also dropping the CSSOM-View IDL afterwards?

Presumably we want to end up with these events defined in one place, not two. I don't have strong opinions about how to achieve that.

> A quick note regarding the past discussion here about Blink (I just saw this discussion)

If that's what you need for compat, it sounds like you should propose changes to the specs so that they match your implementation.

**Comment by mustaqahmed** [comment]
I agree with @jgraham: that's the way forward to avoid more confusions.  I will now comment in https://github.com/w3c/csswg-drafts/issues/4084 about the change we would like to see in this space.
